chore(doc-gen, docs-app): generate version for stable snapshot

Narretz opened the pull request:
The "stable snapshot" is based on the latest commit on the current stable
branch, i.e. a preview of the next patch version.

Next thing:
- update the Firebase deployment so that on the branch with distTag = latest gets deployed to docs.angularjs.org and as to gcs as snapshot-stable

Narretz committed:
chore(doc-gen, docs-app): generate "stable snapshot" for distTag=latest

The "stable snapshot" is the current state of the branch that has
distTag=latest, i.e. a preview of the next patch version of the stable branch.

petebacondarwin commented on docs/app/src/versions.js:
```diff
@@ -12,10 +12,16 @@ angular.module('versions', ['currentVersionData', 'allVersionsData'])
+      var versionStr = CURRENT_NG_VERSION.version;
+
+      if (CURRENT_NG_VERSION.isSnapshot) {
+        versionStr = CURRENT_NG_VERSION.distTag === 'latest' ? 'snapshot-stable' : 'snapshot';
```
perhaps we could rename `snapshot` and `snapshot-stable` to `next` and `stable` respectively?

Narretz replied:
I don't know. I think it's important that these are snapshots of a branch. "Stable" says "release" to me, not snapshot. "Next" is better, but it also doesn't say it's a snapshot. What about "snapshot-next"?

Narretz replied:
btw, these are not the labels that are shown - for "snapshot" we show "master", and for "snapshot-stable" we show "v1.x.y+1-snapshot"

petebacondarwin replied:
I see. Cool to leave them as-is then.

petebacondarwin replied:
or `master-snapshot` and `stable-snapshot`

Narretz replied:
The labels in the version picker are now "master-snapshot" and e.g. "v1.6.7-snapshot"

petebacondarwin commented on lib/versions/version-info.js:
```diff
@@ -231,3 +232,4 @@ if (NO_REMOTE_REQUESTS) {
 
 console.log('CDN version (' + versionSource + '):', cdnVersion ? cdnVersion.raw : 'No version found.');
 console.log('Current version (' + versionSource + '):', exports.currentVersion.raw);
+console.log('Latest Stable', exports.latestStable);
```
`exports.latestStable` is never assigned in this file....

petebacondarwin commented on docs/config/processors/versions-data.js:
```diff
@@ -112,6 +129,20 @@ module.exports = function generateVersionDocProcessor(gitData) {
+
+        var label = 'v' + (version.slice(0, -1) + increment) + '-snapshot';
+
+        return label;
```
would it not be safer to use semver to do this massaging of the version?

Narretz committed:
chore(travis): deploy to docs and code when distTag=latest

We now deploy to code.angularjs.org and docs.angularjs.org
when we are on the branch which has distTag=latest set in the
package.json, i.e. the stable branch.

Previously, we deployed to docs only when distTag=latest and
the commit was tagged, and to code only on the master branch.

petebacondarwin reviewed (approved):
Great!
It is a shame that we have the same logic to compute the `snapshot` and `snapshot-stable` version strings repeated in both the dgeni processor and the application code. But I think we can live with that.

gkalpak commented on docs/config/processors/versions-data.js:
```diff
@@ -112,6 +129,20 @@ module.exports = function generateVersionDocProcessor(gitData) {
+      function createSnapshotStableLabel(version) {
+        var semverObj = semver.parse(version);
+
+        var label = 'v' + semverObj.inc('patch').version + '-snapshot';
```
It would be nice if we showed `v1.6.x-snapshot` (instead of `v1.6.7-snapshot`, which is kind of unclear).

Narretz replied:
Yeah, I guess that's clearer
